lib/document/Document.js

```diff
@@ -195,7 +201,17 @@ class Document {
    * @return {Document}
    */
   set(path, value) {
-    lodashSet(this.data, path, value);
+    let encodedValue = value;
+
+    if (Object.keys(this.encodedProperties).includes(path)) {
```

Comment:
That's not so simple. What if a binary field is `one.two.three`, but we set an object `{ there: new Buffer() }` with path `one.two` or even an array of objects?

Reply:
@shumkov `getEncodedProperties` return an object with an exact paths, for example (in case of arrays): `one.two[1].three`

Reply:
Yes, but we could set an object that contains those encoded properties, so we have to update them too. I've explained the implementation to @shuplenkov